examples/fleetautoscaler.yaml

```diff
@@ -0,0 +1,39 @@
+# Copyright 2018 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#
+# Full example of a FleetAutoScaler - this is used to scale a Fleet
+# automatically depending on load
+#
+
+apiVersion: "stable.agones.dev/v1alpha1"
+kind: FleetAutoScaler
+metadata:
+  # FleetAutoScaler Metadata
+  # https://v1-9.docs.kubernetes.io/docs/reference/generated/kubernetes-api/v1.9/#objectmeta-v1-meta
+  Name: fleet-autoscaler-example
+spec:
+  # The name of the fleet to attach to and control. Must be an existing Fleet in the same namespace
+  # as this FleetAutoScaler
+  fleetName: fleet-example
+  # Size of a buffer of "ready" game server instances
+  # The FleetAutoScaler will scale the fleet up and down trying to maintain this buffer, as
+  # instances are being allocated or terminated
+  bufferSize: 5
```

Comment:
Looking at our original [design ticket](https://github.com/GoogleCloudPlatform/agones/issues/334#issuecomment-417785367) we wanted to have `buffer` config with a `strategy` definition (not got huge feeling on `replicaBuffer vs bufferSize vs `size if it comes under the `buffer` key)

If we do that, then that should resolve @EricFortin 's issues re: naming of the CRD - since the original design for this CRD was meant to be extensible, so we didn't have a CRD+controller for all fleet autoscaler strategies.

WDYT? Then once this is complete, we could add a `strategy: custom` strategy after the fact -- which to be honest, I think needs it's own design ticket outside of this implementation and #334 (we could start that now, there's lots of prior art there too we could leverage).

How does everyone feel about that?

Reply:
Sounds good to me.